kube-runtime/src/controller/mod.rs

```diff
@@ -791,4 +869,66 @@ mod tests {
             ),
         );
     }
+
+    #[tokio::test]
+    async fn applier_must_not_deadlock_if_reschedule_buffer_fills() {
+        // This tests that `applier` handles reschedule queue backpressure correctly, by trying to flood it with no-op reconciles
+        // This is intended to avoid regressing on https://github.com/kube-rs/kube-rs/issues/926
+
+        // Assume that we can keep APPLIER_REQUEUE_BUF_SIZE flooded if we have 100x the number of objects "in rotation"
+        // On my (@teozkr)'s 3900X I can reliably trigger this with 10x, but let's have some safety margin to avoid false negatives
+        let items = APPLIER_REQUEUE_BUF_SIZE * 50;
+        // Assume that everything's OK if we can reconcile every object 3 times on average
+        let reconciles = items * 3;
+
+        let (queue_tx, queue_rx) = futures::channel::mpsc::unbounded::<ObjectRef<ConfigMap>>();
+        let (store_rx, mut store_tx) = reflector::store();
+        let applier = applier(
+            |obj, _| {
+                Box::pin(async move {
+                    // Try to flood the rescheduling buffer buffer by just putting it back in the queue immediately
+                    println!("reconciling {:?}", obj.metadata.name);
+                    Ok(Action::requeue(Duration::ZERO))
+                })
+            },
+            |_: &Infallible, _| todo!(),
+            Arc::new(()),
+            store_rx,
+            queue_rx.map(Result::<_, Infallible>::Ok),
+        );
```

Comment:
holy shit, you managed to write a unit test using `applier`! nice.

Reply:
i assume this is a failing test for bounded queues. did it deadlock a lot before you got the postreconciler stuff going?

Reply:
> holy shit, you managed to write a unit test using applier! nice.

And this is why `applier` itself is sans-io :D

> i assume this is a failing test for bounded queues. did it deadlock a lot before you got the postreconciler stuff going?

Yep, this is why the PR is structured as revert workaround, add test, fix bug properly. If you checkout the test commit then it should deadlock reliably.